x/concentrated-liquidity/tick.go

```diff
@@ -183,6 +184,86 @@ func GetMinAndMaxTicksFromExponentAtPriceOne(exponentAtPriceOne sdk.Int) (minTic
 	return math.GetMinAndMaxTicksFromExponentAtPriceOneInternal(exponentAtPriceOne)
 }
 
+// GetTickLiquidityForRangeInBatches returns an array of liquidity depth within the given range of lower tick and upper tick.
+func (k Keeper) GetTickLiquidityForRange(ctx sdk.Context, poolId uint64, lowerTick, upperTick int64) (sdk.Dec, error) {
+	store := ctx.KVStore(k.storeKey)
+
+	// sanity check that pool exists and upper tick is greater than lower tick
+	if !k.poolExists(ctx, poolId) {
+		return sdk.Dec{}, types.PoolNotFoundError{PoolId: poolId}
+	}
+	if upperTick < lowerTick {
+		return sdk.Dec{}, types.InvalidLowerUpperTickError{LowerTick: lowerTick, UpperTick: upperTick}
+	}
+
+	// use false for zeroForOne since we're going from lower tick -> upper tick
+	zeroForOne := false
+	swapStrategy := swapstrategy.New(zeroForOne, sdk.ZeroDec(), k.storeKey, sdk.ZeroDec())
+
+	// get min and max tick for the pool
+	p, err := k.getPoolById(ctx, poolId)
+	if err != nil {
+		return sdk.Dec{}, err
+	}
+	exponentAtPriceOne := p.GetPrecisionFactorAtPriceOne()
+	minTick, maxTick := math.GetMinAndMaxTicksFromExponentAtPriceOneInternal(exponentAtPriceOne)
+
+	// set current tick to min tick, and find the first initialized tick starting from min tick -1.
+	// we do -1 to make min tick inclusive.
+	currentTick := minTick - 1
+
+	nextTick, ok := swapStrategy.NextInitializedTick(ctx, poolId, currentTick)
+	if !ok {
+		return sdk.Dec{}, types.InvalidTickError{Tick: currentTick, IsLower: false, MinTick: minTick, MaxTick: maxTick}
+	}
+	tick, err := k.GetTickByTickIndex(ctx, poolId, nextTick)
+	if err != nil {
+		return sdk.Dec{}, err
+	}
+
+	// if the second tick initialized is greater than the given upper tick, return first tick's liquidity net
+	if nextTick.Int64() > upperTick {
+		return tick.LiquidityNet, nil
+	}
+	// use the smallest tick initialized as the starting point for calculating liquidity.
+	currentLiquidity := tick.LiquidityNet
+	currentTick = nextTick.Int64()
+
+	totalLiquidityWithinRange := currentLiquidity
+	for currentTick <= upperTick {
```

Comment:
I think that instead of looping here and calling `swapStrategy.NextInitializedTick`, we can iterate the tick prefix for that pool using this:
https://github.com/osmosis-labs/osmosis/blob/f447c87e11d024b342dc8cb3db1e51d3f757105d/osmoutils/store_helper.go#L101-L108

I haven't looked deep enough into it so please proceed with caution on this suggestion. However, if we get the iterator to work, it should be considerably more performant than looping

Reply:
Would you mind if we create an issue this for further improvements? I have took a look at it, seems like this implementation would take more time than the current implmenetation, while I do not want to block front end for this query time wise. 

Some of the implementation I was working on: https://github.com/osmosis-labs/osmosis/compare/mattverse/use-iterator?expand=1 
Mainly more time consuming because we need to utilize both key + value used, and we need to parse the key as well to receive the tick index for the key, while we do not have enough helper methods to do this right away

Reply:
Agreed and approved, I trust you'll either make an issue or PR a follow up change 👍